### PR TITLE
[15.0][IMP] account_payment_order: Add Default Tranfer Move Date configuration

### PR DESCRIPTION
**Pull request by victoralmau**
Add Default Tranfer Move Date configuration

Options:
- **Payment date**: Set the payment date in the transfer entry (similar to the existing so far).
- **Now + Payment due date**: Set now date the payment date in the transfer entry and t he corresponding due date in journal items.

ejemplo

Please @pedrobaeza  @carlosdauden and @sergio-teruel can you review it?

@Tecnativa TT49582

### CODE REVIEW

**Review by sergio-teruel — APPROVED** [review]

File: account_payment_order/models/account_payment_mode.py [hunk 1]
```diff
@@ -56,6 +56,14 @@ class AccountPaymentMode(models.Model):
         ],
         string="Default Payment Execution Date",
     )
+    default_transfer_move_date_prefered = fields.Selection(
+        selection=[
+            ("payment_date", "Payment date"),
+            ("now_date_maturity", "Now + Payment due date"),
+        ],
+        default="payment_date",
+        string="Default Tranfer Move Date",
```
Comment (pedrobaeza) [inline account_payment_order/models/account_payment_mode.py]:
```suggestion
        string="Transfer moves dates",
```

File: account_payment_order/models/account_payment_mode.py [hunk 2]
```diff
@@ -56,6 +56,14 @@ class AccountPaymentMode(models.Model):
         ],
         string="Default Payment Execution Date",
     )
+    default_transfer_move_date_prefered = fields.Selection(
```
Comment (pedrobaeza) [inline account_payment_order/models/account_payment_mode.py]: `default_` prefix is for other kind of fields.

```suggestion
    transfer_date_option = fields.Selection(
```

File: account_payment_order/models/account_payment_mode.py [hunk 3]
```diff
@@ -56,6 +56,14 @@ class AccountPaymentMode(models.Model):
...
+        selection=[
+            ("payment_date", "Payment date"),
+            ("now_date_maturity", "Now + Payment due date"),
```
Comment (pedrobaeza) [inline account_payment_order/models/account_payment_mode.py]:
```suggestion
            ("now_date_maturity", "Now + Payment date as due date"),
```

File: account_payment_order/models/account_payment.py [hunk 4]
```diff
@@ -47,3 +48,17 @@ def _compute_payment_method_line_fields(self):
...
+        to a payment order and a specific configuration.
+        """
+        vals_list = super()._prepare_move_line_default_vals(
+            write_off_line_vals=write_off_line_vals
+        )
+        if self.payment_order_id and self.payment_line_ids:
+            payment_mode = self.payment_order_id.payment_mode_id
+            if payment_mode.default_transfer_move_date_prefered == "now_date_maturity":
+                for vals in vals_list:
+                    vals["date_maturity"] = self.payment_line_ids[0].ml_maturity_date
```
Comment (pedrobaeza) [inline account_payment_order/models/account_payment.py]: It's the payment date, not the maturity date:

```suggestion
                    vals["date_maturity"] = self.payment_line_ids[0].date
```

File: account_payment_order/models/account_payment.py [hunk 5]
```diff
@@ -47,3 +48,17 @@ def _compute_payment_method_line_fields(self):
...
         return res
+
+    def _prepare_move_line_default_vals(self, write_off_line_vals=None):
+        """Overwrite date_maturity of the move_lines that are generated when related
+        to a payment order and a specific configuration.
+        """
+        vals_list = super()._prepare_move_line_default_vals(
+            write_off_line_vals=write_off_line_vals
+        )
+        if self.payment_order_id and self.payment_line_ids:
```
Comment (pedrobaeza) [inline account_payment_order/models/account_payment.py]: No need of this `if`

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by victoralmau** [comment]
Changes done.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge minor

**Comment by pedrobaeza** [comment]
Please fw-port it to v16

**Pull request merged**